File: LC875_KokoEatingBananas/solution.go
package lc875kokoeatingbananas

func minEatingSpeed(piles []int, h int) int {
	possible := func(speed int) bool {
		var count int
		for _, pile := range piles {
			count += (pile + speed - 1) / speed
		}
		return count <= h
	}

	low, high := 1, 1000000000
	// instead of finding max or sorting the list we use max constrant as with binary search it take couple of steps to reach the small number
	for low < high {
		mid := low + (high-low)/2
		if possible(mid) {
			high = mid
		} else {
			low = mid + 1
		}
	}
	return low
}

File: LC875_KokoEatingBananas/solution.py
from typing import List
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        def possible(speed):
            return sum((p+speed-1)//speed for p in piles) <= h
        low,high = 1,10 ** 9
        while low <= high:
            mid= low+(high-low)//2
            if possible(mid):
                high=mid-1
            else:
                low=mid+1
        return low